src/main/java/org/sunbird/catalog/controller/CatalogController.java

```diff
@@ -46,4 +46,11 @@ public ResponseEntity<SBApiResponse> createSubSector(@RequestBody Map<String, Ob
 		SBApiResponse response = catalogService.createSubSector(request);
 		return new ResponseEntity<>(response, response.getResponseCode());
 	}
+
+	@PostMapping("/designation/create")
```

Comment:
this implementation is in cb-ext --- just to reuse the framework capability -- but nothing is reused from existing code.... 

after creating term object - you are still calling pores service to update designation... 

instead - just update the endpoints (or add new one) - in pores service to create / update -- that should create a term object - if term creation fails it returns... if term is created then updates the db...